libs/gi/ui/src/hooks/useNumWorkers.tsx

```diff
@@ -0,0 +1,28 @@
+import { clamp } from '@genshin-optimizer/common/util'
+import { useCallback, useEffect, useState } from 'react'
+const LS_KEY = 'num_opt_workers'
+export function useNumWorkers() {
```

Comment:
Could hoist this into common/ui, so that SRO can use it, however, SRO might have different worker requirements.

Reply:
I can move it later if needed, here is good for now